src/derivatives.jl

```diff
@@ -1,16 +1,11 @@
-function derivative(A, t, order = 1)
+function derivative(A, t, order = 1; iguess = A.iguesser)
```

Comment:
why do we need `iguess` as a kwarg here?

Reply:
To support a user-supplied `iguess`:

https://github.com/SouthEndMusic/DataInterpolations.jl/blob/3debd2909676a0c315ee41a76b90fe9ab146d6e8/src/DataInterpolations.jl#L25-L27

Reply:
Ah wait no that's only for interpolation, shouldn't we support that for derivatives too like this?

Reply:
But in that case it should be documented

Reply:
As the time steps is the same, we should just use iguesser from interpolation objects.

Reply:
I mean that we could support an integer iguess from the user that overrules `iguesser` because they have a very fast way to determine it, or do you not think that that is worth it?

Reply:
I don't think that would be worth it. IMO, users should not give `iguess` themselves for each derivative/interpolation computation (not the construction).

Reply:
Ok, but you want to keep https://github.com/SouthEndMusic/DataInterpolations.jl/blob/3debd2909676a0c315ee41a76b90fe9ab146d6e8/src/DataInterpolations.jl#L25-L27 (as it is already there)?

Reply:
I think we can remove it as it is not documented.

Reply:
Ok, thanks for taking the time to discuss this 👍

Reply:
Turns out it is not tested either